src/sage/interfaces/latte.py

```diff
@@ -216,6 +268,107 @@ def count(arg, ehrhart_polynomial=False, multivariate_generating_function=False,
             return Integer(ans)
 
 
+def str_to_multivariate_generating_function(raw_output_str, name=None, **kwds):
+    r"""
+    Helper function for :func:`count` if ``multivariate_generating_function`` is set to ``True``
+    which preprocess the raw output string to a tuple of summands.
+
+    TESTS:
+
+        sage: from sage.interfaces.latte import count, str_to_multivariate_generating_function # optional - latte_int
+        sage: P = Polyhedron(ieqs=[(0, 1, 0, 0), (0, -1, 1, 0)], eqns=[(0, -1, -1, 2)])
+        sage: cddin = P.cdd_Hrepresentation()
+        sage: raw_output_str = count(cddin, cdd=True, raw_output=True, multivariate_generating_function=True) # optional - latte_int
+        sage: str_to_multivariate_generating_function(raw_output_str, name='xi') # optional - latte_int
+        (1 * (-xi0*xi1*xi2 + 1)^-1 * (-xi1^2*xi2 + 1)^-1,)
+    """
+    from sage.rings.integer_ring import ZZ
+    from sage.rings.polynomial.laurent_polynomial_ring import LaurentPolynomialRing
+    import re
+
+    indices_regex = re.compile(r'(?<=\[)(\d*)(?=\])')
+    indices = range(max(ZZ(index) for index in indices_regex.findall(raw_output_str)) + 1)
+    if name is None:
+        name = 'y'
+
+    B = LaurentPolynomialRing(ZZ,
+                        tuple(name + str(k) for k in indices),
+                        len(indices))
+    raw_output_list = raw_output_str[:-1].split('\n + ')
+    return tuple(_str_to_multivariate_generating_function(a, B, **kwds) for a in raw_output_list)
+
+
+def _str_to_multivariate_generating_function(summand, B=None,
+                                             Factorization_sort=False, Factorization_simplify=False,
+                                             sort_factors=False):
+    r"""
+    Helper function for :func:`str_to_multivariate_generating_function`
+    which convert a summand string to a ``Factorization`` object.
+
+    Each summand is in the format
+    .. MATH::
+
+        (\sum_{k\in K}y^{s_0}*y^{v_k}})\prod_{j\in J}(1 - y^{v_j})^{-1}.
+
+    TESTS::
+
+        sage: from sage.interfaces.latte import _str_to_multivariate_generating_function # optional - latte_int
```

Comment:
I think these lines won't need the `# optional`

Reply:
Yes, you are right